metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1112,35 +1102,12 @@ private void appendRaftEvent(String name, Runnable runnable) {
                 if (this != metaLogListener) {
                     log.debug("Ignoring {} raft event from an old registration", name);
                 } else {
-                    try {
-                        runnable.run();
-                    } finally {
-                        maybeCompleteAuthorizerInitialLoad();
-                    }
+                    runnable.run();
                 }
             });
         }
     }
 
-    private void maybeCompleteAuthorizerInitialLoad() {
```

Comment:
The equivalent logic to this is handled by MetadataLoader, right? (not publishing until we've reached the HWM)

Reply:
Yes. I added a comment to the call to `completeInitialLoad`

```
        if (!completedInitialLoad) {
          // If we are receiving this onMetadataUpdate call, that means the MetadataLoader has
          // loaded up to the local high water mark. So we complete the initial load, enabling
          // the authorizer.
          completedInitialLoad = true
          authorizer.completeInitialLoad()
        }
```